src/lib389/cli/dscontainer

```diff
@@ -352,6 +355,19 @@ binddn = cn=Directory Manager
 
     log.info("389-ds-container started.")
 
+    # If oneshot parameter is passed, run an optional script and
+    # shutdown the server.
+    if args.oneshot != False:
+        if args.oneshot != None:
+            if os.path.exists(args.oneshot):
+                log.info("Running oneshot script...")
+                oneshot_p = subprocess.Popen(args.oneshot, stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE, env=os.environ.copy())
+                oneshot_p.wait()
```

Comment:
Is it worth displaying the return code from the oneshot script here?

Reply:
Yes, good idea, I will add it.

Reply:
In addition to exit code I've added script output for debugging purposes, please review.